biblib/tests/unit_tests/test_manage.py

```diff
@@ -197,5 +176,300 @@ def test_delete_stale_users(self):
+                            document_data=self.stub_library.document_view_post_data('add')
+                        )
+                    self.assertEqual(output.get("number_added"), len(self.stub_library.bibcode))
+                    #Check that the document is in the library
+                    library = session.query(Library).filter(Library.id == library_1_id).all()
+                    for _lib in library:
+                        self.assertIn(list(self.stub_library.bibcode.keys())[0], _lib.bibcode)
+
+                #Add a different document to the library
+                with MockSolrQueryService(canonical_bibcode = self.stub_library_2.document_view_post_data('add').get('bibcode')):
+                    output = self.document_view.add_document_to_library(
+                        library_id=library_1_id,
+                        document_data=self.stub_library_2.document_view_post_data('add')
+                    )
+                    self.assertEqual(output.get("number_added"), len(self.stub_library_2.bibcode))
+                    library = session.query(Library).filter(Library.id == library_1_id).all()
+                    for _lib in library:
+                        self.assertIn(list(self.stub_library_2.bibcode.keys())[0], _lib.bibcode)
+
+                with MockSolrQueryService(canonical_bibcode = self.stub_library_3.document_view_post_data('add').get('bibcode')):
+                    output = self.document_view.add_document_to_library(
+                        library_id=library_1_id,
+                        document_data=self.stub_library_3.document_view_post_data('add')
+                    )
+                    self.assertEqual(output.get("number_added"), len(self.stub_library_3.bibcode))
+                    #Check that the document is in the library
+                    library = session.query(Library).filter(Library.id == library_1_id).all()
+                    for _lib in library:
+                        self.assertIn(list(self.stub_library_3.bibcode.keys())[0], _lib.bibcode)
+
+                    service_user = user_1_id
+                    permissions = session.query(Permissions).filter(Permissions.user_id == service_user).all()
+                    libraries = [session.query(Library).filter(Library.id == permission.library_id).one() for permission in permissions if permission.permissions['owner']]
+                    LibraryVersion = sqlalchemy_continuum.version_class(Library)
+                    revision_lengths = []
+                    for library in libraries:
+                        revisions = session.query(LibraryVersion).filter_by(id=library.id).all()
+                        revision_lengths.append(len(revisions))
+                
+                #Now run the obsolete deletion acting as if we are 1 year in the future.
+                current_offset = datetime.now() + relativedelta(years=1)
+                with freezegun.freeze_time(current_offset):
```

Comment:
Nice package find - this is neat

Reply:
Thank you! I really wasn't sure how to go about time testing otherwise.